### PR TITLE
File descriptors handling

### PR DESCRIPTION
**Pull request by spoutn1k**
Copied the `subprocess.Popen` API, with `close_fds` and `pass_fds` flags at child creation. For #54.

### CODE REVIEW

File: ptrace/debugger/child.py [hunk 1]
```diff
@@ -87,21 +88,28 @@ def _createParent(pid, errpipe_read):
...
-            pass
+    if close_fds:
+        # Close all files except 0, 1, 2 and errpipe_write
+        for fd in range(3, MAXFD):
+            if fd == errpipe_write or (fd in pass_fds):
+                continue
+            try:
+                close(fd)
+            except OSError:
+                pass
```
Comment (vstinner) [inline ptrace/debugger/child.py:111]: pass_fds file descriptors must be made inheritable with os.set_inheritable(). By default, Python creates non-inheritable FDs.

File: ptrace/debugger/child.py [hunk 2]
```diff
@@ -143,6 +151,12 @@ def createChild(arguments, no_stdout, env=None):
      - env={} to start with an empty environment
      - env=None (default) to copy the environment
     """
+
+    if not RUNNING_WINDOWS:
+        if pass_fds and not close_fds:
+            warning("pass_fds overriding close_fds.")
+            close_fds = True
```
Comment (vstinner) [inline ptrace/debugger/child.py:158]: I don't see why you must force close_fds=True here.

Comment (spoutn1k) [reply ptrace/debugger/child.py:158]: I followed the python API for `subprocess.Popen`, where setting a value to `pass_fds` implies closing the other file descriptors.

From the official doc:
> pass_fds is an optional sequence of file descriptors to keep open between the parent and child. Providing any pass_fds forces close_fds to be True. (POSIX only)

File: ptrace/debugger/child.py (hunk 1 above)
Comment (spoutn1k) [reply ptrace/debugger/child.py:114]: I am confused to see `_posixsubprocess` does that, as I manually set that flag on my python-opened file descriptors.
Pushed a new commit for that.

File: ptrace/debugger/child.py [hunk 3]
```diff
@@ -159,6 +159,8 @@ def createChild(arguments, no_stdout, env=None, close_fds=True, pass_fds=()):
         if pass_fds and not close_fds:
             warning("pass_fds overriding close_fds.")
             close_fds = True
+    else:
+        assert not pass_fds, "pass_fds not supported on Windows."
```
Comment (spoutn1k) [inline ptrace/debugger/child.py:163]: I also ripped that from `Lib/subprocess.py`, hope that is ok

File: ptrace/debugger/child.py (hunk 2 above)
Comment (vstinner) [reply ptrace/debugger/child.py:161]: Ok. Please remove the warning.

File: ptrace/debugger/child.py (hunk 3 above)
Comment (vstinner) [reply ptrace/debugger/child.py:163]: python-ptrace is not supported on Windows. Please remove "if not RUNNING_WINDOWS:" and this code.

Comment (spoutn1k) [reply ptrace/debugger/child.py]: Done. I got confused by the RUNNING_WINDOWS flag, that I used but did not implement. Pushed requested changes.

**Comment by vstinner** [comment]
Merged, thanks.

**Pull request merged**